src/script/api/script_town.cpp

```diff
@@ -288,7 +288,7 @@
 	EnforcePrecondition(false, size == TOWN_SIZE_SMALL || size == TOWN_SIZE_MEDIUM || size == TOWN_SIZE_LARGE)
 	EnforcePrecondition(false, size != TOWN_SIZE_LARGE || ScriptObject::GetCompany() == OWNER_DEITY);
 	if (ScriptObject::GetCompany() == OWNER_DEITY || _settings_game.economy.found_town == TF_CUSTOM_LAYOUT) {
-		EnforcePrecondition(false, layout == ROAD_LAYOUT_ORIGINAL || layout == ROAD_LAYOUT_BETTER_ROADS || layout == ROAD_LAYOUT_2x2 || layout == ROAD_LAYOUT_3x3);
+		EnforcePrecondition(false, layout == ROAD_LAYOUT_ORIGINAL || layout == ROAD_LAYOUT_BETTER_ROADS || layout == ROAD_LAYOUT_2x2 || layout == ROAD_LAYOUT_3x3 || layout == ROAD_LAYOUT_RANDOM);
```

Comment:
Is this precondition necessary now that any layout is valid? Or is it to prevent against invalid layouts being passed?

Reply:
It prevents invalid layouts being passed

Reply:
In that case, I wonder if it might be better to check `layout >= ROAD_LAYOUT_BEGIN && layout < ROAD_LAYOUT_END`, obviously after creating those from the TownLayout enum.

Reply:
I can do layout >= ROAD_LAYOUT_ORIGINAL and layout <= ROAD_LAYOUT_RANDOM. Never seen a Script enum with a BEGIN and END listed in them.